job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -273,6 +273,33 @@ class WebApi(system: ActorSystem,
               }
             }
           }
+        } ~
+        put {
+          parameter("reset") { reset =>
+            respondWithMediaType(MediaTypes.`text/plain`) { ctx =>
+              reset match {
+                case "all" => {
+                  import ContextSupervisor._
+                  import collection.JavaConverters._
+                  logger.warn("refreshing contexts")
+                  val future = (supervisor ? ListContexts).mapTo[Seq[String]]
+                  Await.result(future, 1 second).asInstanceOf[Seq[String]]
+                    .foreach { context => {
```

Comment:
Two thoughts: you should parallelize the futures here, and log each one.  Something like this:

    val stopFutures = contexts.map(c => supervisor ? StopContext(c))
    Await.ready(Future.sequence(stopFutures), 15 seconds)

Also I would advise making the timeout configurable.